### ISSUE TITLE
[Advanced Tutorial] Route Planning: The Journey is the Destination

### ISSUE THREAD

**Issue by EVGrokker** [aesculus/EVTO-App-Feedback#324]
this is a placeholder

**Comment by aesculus** [comment]
See my comments via email. Overall I really like it but it does not show auto optimization. In fact this was the way I originally built the app to be be used and many early people felt it sucked because it did not do everything for them. :-(

**Comment by aesculus** [comment]
BTW I had no errors when planning the trip with EVTO. Worked flawlessly.

**Comment by EVGrokker** [comment]
How would you recommend it be changed to better show auto optimization?

**Comment by aesculus** [comment]
In the email I stated making it a three day trip. Go further south for the second night. Then use the third day to pound it home via superchargers

**Comment by aesculus** [comment]
Boy. Looking at the lack of SC in the PNW I am not sure you could do my suggestion. If you bagged the charger at MoClips then maybe a 2 day trip but I think you might find a flow with the current logic not reserving energy correctly at the first day stop.

Is there something worth while down by Oakville to stop at?

**Comment by EVGrokker** [comment]
Take a look at #267 for a tutorial using auto optimization.

**Comment by aesculus** [comment]
Yeah. So up front you should state that the trip you are defining will not use the automatic charger feature. I am OK with that. It's actually a cleaner way to do these types of trips IMHO.

If you do that your trip is great. Tell them for the auto trip sample to see #267 

BTW when I tried to modify the trip to go to Oakville if force me to charge in and out of Aberdeen.

**Comment by aesculus** [comment]
BTW notice when you do the trip incrementally like this you never see that nasty white destination pin. Every time you make a change the app is recalculating in the background. You may end up with a yellow or red pin, and that would tell you to optimize.

**Comment by EVGrokker** [comment]
Revisions made reflecting your email comments. Please review. Not quite ready publication, but close.

**Comment by EVGrokker** [comment]
>Car Data: X90D, Charger: 48, Wheels: 20, Tires: 0, Pano: 0, 5 Seat: 1, Rear CC: 0, Bat Life: 100 
Device Info: iPhone; CPU iPhone OS 10_3_2 like Mac OS X 
Settings: Reserve SoC:20, Default SoC: 90, Region: 0, Units: 0 
Version: 1.2.0 (19) 
IAP: Pro:1, Sub:1 
Location: Unknown 
Trip: Sand Dollar Motel, [317lgae66vs8yob00urpdi]

Hangs during optimization.

**Comment by aesculus** [comment]
Yes it does. Sheesh. On it.

**Comment by aesculus** [comment]
Can you go to testflight and change your release to (17) and see if the problem is there too? That will help me figure out if its on the app or server.

**Comment by EVGrokker** [comment]
Hangs under (17) as well. 

G

**Comment by aesculus** [comment]
OK. I flummoxed something on the server working on 18-19.  It may take me awhile to figure out what.

**Comment by aesculus** [comment]
Try this with V1.2 (20)

**Comment by aesculus** [comment]
Can you try this again with 1.2 (33)?  If it works do you want to deploy now or wait for V1.2?

**Comment by EVGrokker** [comment]
I don't want to deploy any of the not-yet-published tutorials until I have a chance to review them with the new features and UI changes. I definitely want to take advantage of the opportunity to showcase the flashy new features whenever possible. I will try the trip under 33 to see how it behaves.

**Comment by EVGrokker** [comment]
I'm now able to get all the way through the tutorial without being flummoxing, but there's an anomaly and a problem.

>Car Data: X90D, Charger: 48, Wheels: 20, Tires: 0, Pano: 0, 5 Seat: 0, Rear CC: 0, Bat Life: 90 
Device Info: iPhone; CPU iPhone OS 10_3_2 like Mac OS X 
Settings: Reserve SoC:20, Default SoC: 90, Region: 0, Units: 0 
Version: 1.2.0 (35) 
IAP: Pro:1, Sub:1 
Location: Unknown 
Trip: Sand Dollar Inn & Cottages, [yl3hnuh36kdow5f97g14]

The return trip automatically added a charging stop in Aberdeen WA, but there's no charging allocated at the stop - arrival and departure SoC are both 78%. That stop shouldn't be there.

If I Optimize Route, the dialog hangs at Selecting Superchargers.

**Comment by EVGrokker** [comment]
I see that you noted similar problems in issue #414.

**Comment by aesculus** [comment]
I am working on that right now. I thought it was a simple fix but it is perplexing me a bit. I know where the issues stems from so I hope to have it resolved today.

Basically when there is a charger at a hinge, it's not treating it correctly and assumes it needs to charge anyway.

**Comment by aesculus** [comment]
Please try optimizing again to see if Aberdeen is now gone.

**Comment by EVGrokker** [comment]
I tried optimizing the existing trip, but it still hangs.

So I created a new round trip from Seattle to Sand Dollar Inn, and the return leg shows no stop in Aberdeen. So I think you resolved that one. I will try the entire trip again from scratch and see if everything works as expected.

**Comment by aesculus** [comment]
That hang may be a bug that (36) fixes. I did not log it.  It happened when waypoints were deleted. But I did not think it would happen during optimization but maybe ...

**Comment by EVGrokker** [comment]
v 1.2.0 (36)
In recreating this trip, when I get to step 5.1, the origin shows 90% SoC (correct), but the destination shows 0%, which means that the return segment hasn't been since I added the outbound waypoints. The destination also shows 'Invalid date' on the arrival time. As soon as I make an edit to the return segment (changing the departure time), the arrival SoC is now 33%, and the arrival time is now accurate.

From your perspective, is that correct behavior? Or should EVTO be recomputing the downstream segment when changes are made upstream?

The bad news is that after completing the route construction, selecting Optimize Routing from the menu still hangs.
